presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileWriter.java

```diff
@@ -188,11 +188,15 @@ private void writeMetadataProperty(String key, String value)
     public void close()
             throws IOException
     {
-        writeRowGroup();
-        output.close();
-        keySectionOutput.destroy();
-        for (ColumnEncoder columnEncoder : columnEncoders) {
-            columnEncoder.destroy();
+        try {
+            writeRowGroup();
+            output.close();
+        }
+        finally {
+            keySectionOutput.destroy();
+            for (ColumnEncoder columnEncoder : columnEncoders) {
+                columnEncoder.destroy();
```

Comment:
destroy() is going to put the compressors directly back to the pool without throwing an exception. The code of returnCompressor() in both our hadoop and apache hadoop looks safe. Given that, do we wanna wrap try/catch around returnCompressor() just in case (e.g., future code change in hadoop causing a runtime exception)?